models/gc_autopet_fpr/dockerfiles/Dockerfile

```diff
@@ -34,6 +34,16 @@ RUN mkdir -p /opt/algorithm/nnUNet_raw_data_base/nnUNet_raw_data/Task504_Total_P
 ENV nnUNet_raw_data_base="/opt/algorithm/nnUNet_raw_data_base"
 ENV RESULTS_FOLDER="/opt/algorithm/checkpoints"
 
+# install DICOM TCIA preprocessing (registration and nifti conversion)
+RUN git clone https://github.com/lab-midas/TCIA_processing /tmp/algorithm && \
```

Comment:
There is a robust converter from DICOM to NIfTI, which we are already using: dcm2niix. I do not see any reason to experiment with any other tools. I strongly recommend this is taken out and replaced with dcm2niix.

Reply:
This particular processing performs the necessary registration of the CT to the PET scan as well and comes with the data provided on TCIA, I recommend you use this instead of the dcm2nixx tool.

Reply:
Can you point me to the details about that registration? I do not see any mention of "registration" in either README file, or in the code in the repo.

That repo was literally developed for that one specific collection. It is unacceptable for general purpose model to use collection-specific code. The whole point is to make the model robust to any similar dataset. 

I strongly disagree with the suggestion to use that tool. Instead, I recommend you handle CT and PET as two separate channels, similar to how MR T2/ADC are handled in #60.

Reply:
It is true that it isn't mentioned very well in the README, but it's there:
* https://wiki.cancerimagingarchive.net/pages/viewpage.action?pageId=93258287: at the bottom under "Additional Resources for this Dataset" mentions: "Scripts provided by the submitting group for file conversion, preprocessing alignment and resampling of PET, CT and mask data"
* In the repo https://github.com/lab-midas/TCIA_processing:
  * In the README.md the output for the "TCIA_processing: tcia_nifti_to_mha.py" mentions a CTres.nii.gz (CT image resampled to PET image resolution and size) (see the output image)
  * The code for this can be found here: https://github.com/lab-midas/TCIA_processing/blob/fb7b0740bcab896ed8c8e646c4491cadafbb8789/tcia_dicom_to_nifti.py#L139-L144

I am also convinced that for this dataset the preprocessing pipeline should work for other similar data if these were available (currently no other IDC dataset is a good match). The conversion and registration code will likely work if similar data is presented.

Reply:
> @silvandeleemput the code you referenced does not do any registration - it is just the resampling.
> 

My bad. Where I said registration should have been resampling, that's correct.

> > I am also convinced that for this dataset the preprocessing pipeline should work for other similar data if these were available (currently no other IDC dataset is a good match). The conversion and registration code will likely work if similar data is presented.
> 
> I love your optimism, but (just after a very quick look around) I found this line that seems to indicate that the world is not perfect: https://github.com/lab-midas/TCIA_processing/blob/fb7b0740bcab896ed8c8e646c4491cadafbb8789/tcia_dicom_to_nifti.py#L124. And the things they do after that line are just a recipe for a disaster: https://github.com/lab-midas/TCIA_processing/blob/fb7b0740bcab896ed8c8e646c4491cadafbb8789/tcia_dicom_to_nifti.py#L127-L128
> 

This is indeed very bad for generalization to other datasets, good catch! It is of course possible to use the dcm2niix tool first and subsequently only use the resampling part of the TCIA codebase (might not even require the TCIA preprocessing install as it seems to be mostly nifti calls). Only the resampling part is essential to the model. We should probably test if both tools produce the same image data on the TCIA dataset though.

> > (currently no other IDC dataset is a good match)
> 
> Just to set the record straight, there are a lot of collections in IDC that contain PET/CT studies: https://portal.imaging.datacommons.cancer.gov/explore/filters/?Modality_op=AND&Modality=PT&Modality=CT

I agree, but the model was trained on the [(FDG-PET-CT-Lesions)](https://wiki.cancerimagingarchive.net/pages/viewpage.action?pageId=93258287) dataset specifically, the model might not generalize that well to other datasets.

Reply:
> > and subsequently only use the resampling part of the TCIA codebase
> 
> I do not see how it is justified to adopt this random package just for the sake of resampling. Resampling is just a few lines of SimpleITK. Plastimatch has a resampling operator: https://plastimatch.org/plastimatch.html#plastimatch-resample.

If you read the lines below that line carefully, you'll see that I am actually open to removing the dependency on this TCIA library. Indeed resampling is available in other packages, I just want it to be similar to the one used for the model data in the GC challenge to allow more reproducible results. If these other packages resample in the exact the same way I am fine with that too.